src/algokit/core/compile.py

```diff
@@ -0,0 +1,80 @@
+from collections.abc import Iterator
```

Comment:
I'm wondering if we should setup the structure to support additional compilers now?

Similar to what we've done with `task`.
I guess we'd have a directory `compile` inside both `cli` and `core`, which has a `py.py` file? Worth also double confirming that we are happy with `py` and don't want to use `python`?